### ISSUE TITLE
Compile fails on AIX using xlc

### ISSUE THREAD

**Issue by marioroy** [Sereal/Sereal#171]
The following addition to snappy/csnappy_internal_userspace.h resolves the issue.

```
--- csnappy_internal_userspace.h.orig   2016-07-12 19:39:28.000000000 -0500
+++ csnappy_internal_userspace.h        2017-06-07 05:18:56.000000000 -0500
@@ -154,6 +154,12 @@
  #define __LITTLE_ENDIAN LITTLE_ENDIAN
  #define __BIG_ENDIAN BIG_ENDIAN

+ #elif defined(_AIX)
+
+ #include <sys/machine.h>
+ #define __BYTE_ORDER BYTE_ORDER
+ #define __LITTLE_ENDIAN LITTLE_ENDIAN
+ #define __BIG_ENDIAN BIG_ENDIAN

  #elif defined(__sun)

```

Testing passes on an AIX LPAR with the patch applied to csnappy_internal_userspace.h.

```
l48fvp035_pub[/home/u0018708/modules/Sereal-Encoder-3.015] > make test
        PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t t/*/*/*.t
t/001_load.t ............................. ok
t/002_constants.t ........................ ok
t/002_have_enc_and_dec.t ................. # Testing with both encoder and decoder.
# Sereal::Decoder v3.015
# Sereal::Encoder v3.015
t/002_have_enc_and_dec.t ................. ok
t/003_ptable.t ........................... ok
t/004_testset.t .......................... ok
t/005_flags.t ............................ ok
t/010_desperate.t ........................ ok
t/011_aliased_dedupe.t ................... ok
t/020_sort_keys.t ........................ ok
t/021_sort_keys_option.t ................. ok
t/022_canonical_refs.t ................... ok
t/030_canonical_vs_test_deep.t ........... skipped: We are skipping all our tests because we don't have a suitable Test::Deep here, ...
t/040_tied_hash.t ........................ ok
t/110_nobless.t .......................... ok
t/120_hdr_data.t ......................... ok
t/130_freezethaw.t ....................... ok
t/160_recursion.t ........................ ok
t/170_cyclic_weakrefs.t .................. ok
t/180_magic_array.t ...................... ok
t/200_bulk.t ............................. ok
t/300_fail.t ............................. ok
t/400_evil.t ............................. ok
t/700_roundtrip/v1/plain.t ............... ok
t/700_roundtrip/v1/plain_canon.t ......... ok
t/700_roundtrip/v1/snappy.t .............. ok
t/700_roundtrip/v1/snappy_canon.t ........ ok
t/700_roundtrip/v2/dedudep_strings.t ..... ok
t/700_roundtrip/v2/freeze_thaw.t ......... ok
t/700_roundtrip/v2/plain.t ............... ok
t/700_roundtrip/v2/plain_canon.t ......... ok
t/700_roundtrip/v2/readonly.t ............ ok
t/700_roundtrip/v2/snappy.t .............. ok
t/700_roundtrip/v2/snappy_canon.t ........ ok
t/700_roundtrip/v2/snappy_incr.t ......... ok
t/700_roundtrip/v2/snappy_incr_canon.t ... ok
t/700_roundtrip/v2/sort_keys.t ........... ok
t/700_roundtrip/v3/dedudep_strings.t ..... ok
t/700_roundtrip/v3/freeze_thaw.t ......... ok
t/700_roundtrip/v3/plain.t ............... ok
t/700_roundtrip/v3/plain_canon.t ......... ok
t/700_roundtrip/v3/readonly.t ............ ok
t/700_roundtrip/v3/snappy.t .............. ok
t/700_roundtrip/v3/snappy_canon.t ........ ok
t/700_roundtrip/v3/snappy_incr.t ......... ok
t/700_roundtrip/v3/snappy_incr_canon.t ... ok
t/700_roundtrip/v3/sort_keys.t ........... ok
t/700_roundtrip/v3/sort_keys_perl.t ...... ok
t/700_roundtrip/v3/sort_keys_perl_rev.t .. ok
t/700_roundtrip/v3/zlib.t ................ ok
t/700_roundtrip/v3/zlib_force.t .......... ok
t/800_threads.t .......................... ok
t/900_reentrancy.t ....................... ok
All tests successful.
Files=52, Tests=185969, 109 wallclock secs ( 8.59 usr  0.48 sys + 57.30 cusr  0.60 csys = 66.97 CPU)
Result: PASS
Target "test" is up to date.
```

Regards,
Mario

**Comment by ikruglov** [comment]
Hi. Thanks for reporting. This issue was addressed in #169. Commit https://github.com/Sereal/Sereal/pull/169/commits/fd1e66f23ac1174578c83d8deac257ee09183045

**Comment by marioroy** [comment]
Thank you @ikruglov. Recently, had access to an IBM LPAR VM and thought to validate MCE, MCE::Shared, and the two optional modules Sereal and IO::FDPass. IO::FDPass works fine. And Sereal can too. I was not aware of a commit already made. Thank you, @andygrundman.

For serialization, both MCE and MCE::Shared default to Sereal::{ Decoder, Encoder } 3.015 or later if available. Otherwise, defaults to Storable.
